irohad/pending_txs_storage/impl/pending_txs_storage_impl.cpp

```diff
@@ -64,10 +68,29 @@ PendingTransactionStorageImpl::getPendingTransactions(
   while (account_batches.batches.end() != batch_iterator
          and remaining_space >= batch_iterator->get()->transactions().size()) {
     auto &txs = batch_iterator->get()->transactions();
-    response.transactions.insert(
-        response.transactions.end(), txs.begin(), txs.end());
-    remaining_space -= txs.size();
-    ++batch_iterator;
+    // can be easily extended by adding more conditions if needed
+    if (first_tx_time or last_tx_time) {
+      auto time_predicate = [&](auto &tx) {
+        bool first_tx_time_valid =
+            first_tx_time ? first_tx_time < tx->createdTime() : true;
+        bool last_tx_time_valid =
+            last_tx_time ? tx->createdTime() < last_tx_time : true;
+        return first_tx_time_valid && last_tx_time_valid;
+      };
+      auto iter_before_copy = std::end(response.transactions);
+      std::copy_if(txs.begin(),
```

Comment:
And the last *optional* senior question, what about batches, are they used somewhere else after this function, or we can consider past batches as "last usage before removal", so their data (i.e. transactions) could be safely moved instead of copied.

*This is a deep question and not a requirement.*

Reply:
But if transactions are shared pointers, this does not matter. (I did not go deeper in code, just thought)

Reply:
I don't know if we are thinkg about same transactions but in irohad/pending_txs_storage/pending_txs_storage.hpp `response.transactions` are stored as `shared_model::interface::types::SharedTxsCollectionType` which is defined in shared_model/interfaces/common_objects/transaction_sequence_common.hpp as `using SharedTxsCollectionType = std::vector<std::shared_ptr<Transaction>>;` so they are shared pointers